lsp-mode.el

```diff
@@ -8059,6 +8103,7 @@ such."
   (interactive)
   (with-no-warnings (when (functionp 'lsp-ui-mode) (lsp-ui-mode -1)))
   (lsp--text-document-did-close t)
+  (lsp-unconfig-buffer)
```

Comment:
can we call unconfig buffer when disabling lsp-managed-mode? It seems like it should be there. The same applies to setting workspaces to nil. Feel free to merge without applying this comment.

Reply:
Nice, I will include `lsp-configure-buffer` and `lsp-unconfig-buffer` in enable/disable `lsp-managed-mode` respectively.
For setting workspaces to nil, it would be better to set all `defvar-local` variables to nil when disable `lsp-managed-mode`.
That can be another PR.